src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,457 @@
+        final DetailAST parent = identAst.getParent();
+        return isStandAloneIncrementOrDecrement(identAst)
+                || parent.getType() == TokenTypes.ASSIGN
+                && identAst != parent.getLastChild();
+    }
+
+    /**
+     * Checks whether the DetailAst of type {@link TokenTypes#IDENT} is used as
+     * an operand of a stand-alone increment or decrement.
+     *
+     * @param identAst DetailAst of type {@link TokenTypes#IDENT}
+     * @return True if identAst is used as an operand of stand-alone
+     *         increment or decrement otherwise false
+     */
+    private static boolean isStandAloneIncrementOrDecrement(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        final DetailAST grandParent = parent.getParent();
+        return TokenUtil.isOfType(parent, INCREMENT_AND_DECREMENT_TOKENS)
+                && TokenUtil.isOfType(grandParent, TokenTypes.EXPR)
+                && (!TokenUtil.isOfType(grandParent.getParent(),
+                TokenTypes.ELIST, TokenTypes.INDEX_OP, TokenTypes.ASSIGN)
+                || grandParent.getParent().getParent().getType() == TokenTypes.FOR_ITERATOR);
+    }
+
+    /**
+     * Maintains information about the local variable.
+     */
+    private static final class VariableDesc {
+
+        /**
+         * The name of the variable.
+         */
+        private final String name;
+
+        /**
+         * DetailAst of type {@link TokenTypes#TYPE}.
+         */
+        private final DetailAST typeAst;
+
+        /**
+         * The scope of variable is determined by the DetailAst of type
+         * {@link TokenTypes#SLIST} which is enclosing the variable.
+         */
+        private final DetailAST scope;
+
+        /**
+         * Is the variable used again.
+         */
+        private boolean used;
+
+        /**
+         * Create a new VariableDesc instance.
+         *
+         * @param name name of the variable
+         * @param typeAst DetailAst of type {@link TokenTypes#TYPE}
+         * @param scope {@link VariableDesc#scope}
+         */
+        /* package */ VariableDesc(String name, DetailAST typeAst, DetailAST scope) {
+            this.name = name;
+            this.typeAst = typeAst;
+            this.scope = scope;
+        }
+
+        /**
+         * Get the name of variable.
+         *
+         * @return name of variable
+         */
+        public String getName() {
+            return name;
+        }
+
+        /**
+         * Get the associated {@link VariableDesc#typeAst}.
+         *
+         * @return {@link VariableDesc#typeAst}
+         */
+        public DetailAST getTypeAst() {
+            return typeAst;
+        }
+
+        /**
+         * Get the associated {@link VariableDesc#scope}.
+         *
+         * @return {@link VariableDesc#scope}.
+         */
+        public DetailAST getScope() {
+            return scope;
+        }
+
+        /**
+         * Is the variable used again or not.
```

Comment:
`again` - to me, the variable is used or not. `again` sounds like a second usage.

Reply:
removed `again`

Reply:
Resolved.